src/Tuple/index.js

```diff
@@ -0,0 +1,163 @@
+    constant(_type(n))
+
+  const typeString = typeFn('Tuple', VERSION, n)
+
+  const withProps = fn => {
+    fn.type = type
+    fn['@@type'] = typeString
+    fn['@@implements'] = _implements([ 'map', 'concat', 'equals' ])
+    return fn
+  }
+
+  /* eslint-disable no-unused-vars */
+  switch (n) {
+  case 1: return withProps(function(a) { return Tuple(n, arguments) })
+  case 2: return withProps(function(a, b) { return Tuple(n, arguments) })
+  case 3: return withProps(function(a, b, c) { return Tuple(n, arguments) })
+  case 4: return withProps(function(a, b, c, d) { return Tuple(n, arguments) })
+  case 5: return withProps(function(a, b, c, d, e) { return Tuple(n, arguments) })
+  case 6: return withProps(function(a, b, c, d, e, f) { return Tuple(n, arguments) })
+  case 7: return withProps(function(a, b, c, d, e, f, g) { return Tuple(n, arguments) })
+  case 8: return withProps(function(a, b, c, d, e, f, g, h) { return Tuple(n, arguments) })
+  case 9: return withProps(function(a, b, c, d, e, f, g, h, i) { return Tuple(n, arguments) })
+  case 10: return withProps(function(a, b, c, d, e, f, g, h, i, j) { return Tuple(n, arguments) })
+  default:
+    throw new TypeError(
+      'Tuple: Tuple size should be a number between 1 and 10'
+    )
+  }
+  /* eslint-enable no-unused-vars */
+
+  function Tuple(n, args) {
+    const parts = [].slice.call(args)
+    if (n !== parts.length) {
+      throw new TypeError(
+        `${n}-Tuple: Expected ${n} values, but got ${parts.length}`
+      )
+    }
+
+    const inspect =
+      () => `${n}-Tuple(${parts.map(_inspect).join(',')} )`
+
+    function map(method) {
+      return function(fn) {
+        if (!isFunction(fn)) {
+          throw new TypeError(`${n}-Tuple.${method}: Function required`)
+        }
+
+        return Tuple(n, parts
+          .slice(0, parts.length - 1)
+          .concat(fn(parts[parts.length - 1]))
+        )
+      }
+    }
+
+    const equals = m =>
+      isSameType({ type }, m)
+        && _equals(parts, m.toArray())
+
+    function concat(method) {
+      return function(t) {
+        if (!isSameType({ type }, t)) {
+          throw new TypeError(`${n}-Tuple.${method}: Tuple of the same length required`)
+        }
+
+        const a = t.toArray()
+
+        return Tuple(n, parts.map((v, i, o) => {
+          if (!(isSemigroup(a[i]) && isSemigroup(o[i]))) {
+            throw new TypeError(
+              `${n}-Tuple.${method}: Both Tuples must contain Semigroups of the same type`
+            )
+          }
+
+          if (!isSameType(a[i], o[i])) {
+            throw new TypeError(
+              `${n}-Tuple.${method}: Both Tuples must contain Semigroups of the same type`
+            )
+          }
+
+          return o[i].concat(a[i])
+        }))
+      }
+    }
+
+    function merge(fn) {
+      if (!isFunction(fn)) {
+        throw new TypeError(`${n}-Tuple.merge: Function required`)
+      }
+
+      return fn(...parts)
+    }
+
+    function nmap(...args) {
```

Comment:
I thought about this and then I noticed that all other pointfree functions are named after instance methods. I'd naturally reach out for `pointfree/mapAll` if I'm trying to convert a `Tuple(..).mapAll(..)` to a pointfree compose chain.

That being said, I feel your reasoning is valid because `nmap` does take context. The only other way we can call the pointfree function `mapAll` would be to make it smarter by detecting `Tuple` lengths (https://github.com/evilsoft/crocks/pull/275/commits/197c9d1d9a16f0c934c06eb668f0be0d5ce861fd) - but since we decided against that, I'm going to go ahead and make this change.